pkg/types/azure/metadata.go

```diff
@@ -7,6 +7,8 @@ type Metadata struct {
 	Region                      string           `json:"region"`
 	ResourceGroupName           string           `json:"resourceGroupName"`
 	BaseDomainResourceGroupName string           `json:"baseDomainResourceGroupName"`
+	ZoneName                    string           `json:"zoneName"`
+	APIURL                      string           `json:"apiURL"`
```

Comment:
This would work going forward, but we need to be able to keep deleting clusters that were installed before this change went into affect. So we cannot always depend on having zoneName and APIURL.

It seems like it is the best way to delete (get by record name, ensure the record is for the correct cluster via metadata then delete), but if the metadata is missing we can fall back to the old method.

Reply:
> It seems like it is the best way to delete (get by record name, ensure the record is for the correct cluster via metadata then delete), but if the metadata is missing we can fall back to the old method.

Actually I think the way you have it currently structured is fine too: we start with the old method, and if that fails, try the new method. That's fine with me.

But we may just need to add some checks to bail out if the metadata is from an old install and does not have the needed values.